### PR TITLE
Remove QMSI and unused drivers.

### PR DESCRIPTION
**Pull request by nashif**
Remove QMSI and any drivers supporting obsolete platforms.

### CODE REVIEW

**Comment by ulfalizer** [comment]
> Found references to undefined Kconfig symbols. If any of these are false
> positives, then add them to UNDEF_KCONFIG_WHITELIST in check_compliance.py in the ci-tools repo.
> 
> If the reference is for a comment like /* CONFIG_FOO_* */ (or
> /* CONFIG_FOO_*_... */), then please use exactly that form (with the '*'). The
> CI check knows not to flag it.
> 
> More generally, a reference followed by $, @, {, *, or ## will never be
> flagged.
> 
> CONFIG_COUNTER_0_NAME                      soc/arm/atmel_sam0/samd20/dts_fixup.h:9, soc/arm/atmel_sam0/samd21/dts_fixup.h:9, soc/arm/atmel_sam0/samr21/dts_fixup.h:13

Might be a false positive due to naming, just covered up before:

```
$ git grep COUNTER_0_NAME
drivers/counter/Kconfig.qmsi:config COUNTER_0_NAME
drivers/counter/counter_qmsi_aonpt.c:DEVICE_DEFINE(aon_timer, CONFIG_COUNTER_0_NAME, aon_timer_init,
soc/arm/atmel_sam0/samd20/dts_fixup.h:#define CONFIG_COUNTER_0_NAME DT_INST_0_ATMEL_SAM0_TC32_LABEL
soc/arm/atmel_sam0/samd21/dts_fixup.h:#define CONFIG_COUNTER_0_NAME DT_INST_0_ATMEL_SAM0_TC32_LABEL
soc/arm/atmel_sam0/samr21/dts_fixup.h:#define CONFIG_COUNTER_0_NAME DT_INST_0_ATMEL_SAM0_TC32_LABEL
tests/drivers/counter/counter_basic_api/src/test_counter.c:#ifdef CONFIG_COUNTER_0_NAME
tests/drivers/counter/counter_basic_api/src/test_counter.c:     CONFIG_COUNTER_0_NAME,
```

Could make it `#define CFG_COUNTER_0_NAME ...` if you decide to go for some renaming did. That that for some other stuff. Haven't looked into whether it makes sense here though.

**Review by dbkinder — APPROVED** [review]

**Review by galak — APPROVED** [review]

**Review by ioannisg — APPROVED** [review]

File: soc/arm/atmel_sam0/samd20/dts_fixup.h [hunk 1]
```diff
@@ -6,8 +6,6 @@
 
 #define CONFIG_WDT_0_NAME DT_INST_0_ATMEL_SAM0_WATCHDOG_LABEL
 
-#define CONFIG_COUNTER_0_NAME DT_INST_0_ATMEL_SAM0_TC32_LABEL
```
Comment (MaureenHelm) [inline soc/arm/atmel_sam0/samd20/dts_fixup.h]: Why are you modifying atmel socs with this commit?

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

File: soc/arm/atmel_sam0/samd20/dts_fixup.h (hunk 1 above)
Comment (nashif) [reply soc/arm/atmel_sam0/samd20/dts_fixup.h]: CONFIG_COUNTER_0_NAME was only being defined in QMSI drivers, nobody else uses this and CI was complaining

**Comment by nashif** [comment]
@galak can you please add a comment about what changes you forced pushed to this PR?

File: soc/arm/atmel_sam0/samd20/dts_fixup.h (hunk 1 above)
Comment (MaureenHelm) [reply soc/arm/atmel_sam0/samd20/dts_fixup.h]: Will you please note this in the commit message?

Comment (nashif) [reply soc/arm/atmel_sam0/samd20/dts_fixup.h]: done

**Comment by galak** [comment]
> @galak can you please add a comment about what changes you forced pushed to this PR?

was just a rebase to deal with a merge conflict.

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**